libbpf-tools/ttysnoop.bpf.c

```diff
@@ -0,0 +1,145 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (c) 2021 Hengqi Chen */
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "ttysnoop.h"
+
+#define WRITE			1
+/**
+ * starting from linux v5.14-rc1, ITER_IOVEC has value 0 instead of 4
+ * see https://github.com/torvalds/linux/blob/v5.14-rc1/include/linux/uio.h#L22
+ */
+#define ITER_IOVEC_v514		0
+
+/**
+ * starting from linux v5.14-rc1, struct iov_iter has new definition
+ * define our own struct iov_iter since vmlinux.h is outdated
+ * see https://github.com/torvalds/linux/blob/v5.14-rc1/include/linux/uio.h#L30-L50
+ */
+struct iov_iter___v514 {
+	__u8 iter_type;
+	bool data_source;
+	union {
+		const struct kvec *kvec;
+	};
+};
+
+static int zero = 0;
+const volatile __u64 pts = 0;
+const volatile int read_bytes = 256;
```

Comment:
this isn't used in BPF. Maybe it was meant to control `bpf_probe_read_user` ?

Reply:
Yes, as mentioned above, I would like to hear your opinion about this option.